rust/lance/src/dataset/scanner.rs

```diff
@@ -685,18 +692,30 @@ impl Scanner {
     }
 
     /// Create an Execution plan with a scan node
-    fn scan(&self, with_row_id: bool, projection: Arc<Schema>) -> Arc<dyn ExecutionPlan> {
+    fn scan(
+        &self,
+        with_row_id: bool,
+        with_make_deletions_null: bool,
```

Comment:
what there be an implication of this with regular null support? Do we need to differentiate them?

Maybe add some docstring to discuss the considerations?